findoutlie/outfind.py

```diff
@@ -3,13 +3,108 @@
 
 import os.path as op
 from glob import glob
+from functools import partial
 
+import numpy as np
+import nibabel as nib
 
-def detect_outliers(fname):
-    return [42]
+from findoutlie.detectors import std_detector, iqr_detector
+from findoutlie.metrics import dvars
+from findoutlie.spm_funcs import get_spm_globals
+
+def detect_outliers(fname, metric='vol_means', detector='std', 
+                    n_stds=2, iqr_proportion=1.5, mask_join='or'):
+    """
+    Choose a timecourse metric and outlier detection method to apply to a given
+    file.
+
+    Parameters
+    ----------
+    fname : str, the timeseries file to be processed for outliers 
+
+    metric options : can be a str or list of options
+        The metric used for summarizing timecourse data across volumes before
+        applying outlier detecton method.
+
+        Options can include:
+        -'vol_means'
+        -'dvars'
+        -'spm_global'
+        -'all' (chooses all of the above)
+        -list of some or all of the above options
+
+    detector options : can be a str or list of options
+        The outlier detection method to use.
+
+        Options can include:
+        -'std'
+        -'iqr'
+        -'all' (chooses all of the above)
+        -list of some or all of the above options
+    
+    n_stds : int or float (default=2)
+        The standard deviation threshold used with the 'std' 
+        detector method.
+
+    iqr_proportion : int or float (default=1.5)
+        The interquartile proportion threshold to use with the
+        'iqr' method.
+
+    mask_join : str (default='or')
```

Comment:
Or you could actually pass the joining function `np.logical_or`, `np.logical_and` or a custom function.